tests/integration/targets/dhcp_option_space/tasks/main.yml

```
@@ -0,0 +1,181 @@
+        protocol: "ip4"
+        state: present
+      check_mode: true
+      register: option_space
+    - name: Get Information about the DHCP Option Space
+      infoblox.universal_ddi.dhcp_option_space_info:
+        filters:
+          name: "{{ option_space_name }}"
+      register: option_space_info
+    - assert:
+        that:
+          - option_space is changed
+          - option_space_info is not failed
+          - option_space_info.objects | length == 0
+
+    - name: Create a DHCP Option Space
+      infoblox.universal_ddi.dhcp_option_space:
+        name: "{{ option_space_name }}"
+        protocol: "ip4"
+        state: present
+      register: option_space
+    - name: Get Information about the DHCP Option Space
+      infoblox.universal_ddi.dhcp_option_space_info:
+        filters:
+          name: "{{ option_space_name }}"
+      register: option_space_info
+    - assert:
+        that:
+          - option_space is changed
+          - option_space is not failed
+          - option_space_info is not failed
+          - option_space_info.objects | length == 1
+          - option_space_info.objects[0].id == option_space.id
+          - option_space_info.objects[0].name == option_space.object.name
+
+    - name: Create a DHCP Option Space (idempotent)
+      infoblox.universal_ddi.dhcp_option_space:
+        name: "{{ option_space_name }}"
+        protocol: "ip4"
+        state: present
+      register: option_space
+    - assert:
+        that:
+          - option_space is not changed
+          - option_space is not failed
+
+    - name: Delete the DHCP Option Space (check mode)
+      infoblox.universal_ddi.dhcp_option_space:
+        name: "{{ option_space_name }}"
+        state: absent
+      check_mode: true
+      register: option_space
+    - name: Get Information about the DHCP Option Space
+      infoblox.universal_ddi.dhcp_option_space_info:
+        filters:
+          name: "{{ option_space_name }}"
+      register: option_space_info
+    - assert:
+        that:
+          - option_space is changed
+          - option_space_info is not failed
+          - option_space_info.objects | length == 1
+
+    - name: Delete the DHCP Option Space
+      infoblox.universal_ddi.dhcp_option_space:
+        name: "{{ option_space_name }}"
+        state: absent
+      register: option_space
+    - name: Get Information about the DHCP Option Space
+      infoblox.universal_ddi.dhcp_option_space_info:
+        filters:
+          name: "{{ option_space_name }}"
+      register: option_space_info
+    - assert:
+        that:
+          - option_space is changed
+          - option_space_info is not failed
+          - option_space_info.objects | length == 0
+
+    - name: Delete the DHCP Option Space (idempotent)
+      infoblox.universal_ddi.dhcp_option_space:
+        name: "{{ option_space_name }}"
+        state: absent
+      register: option_space
+    - assert:
+        that:
+          - option_space.changed == false
```

This is completely correct , but can we follow one specific format for assertions ?

```suggestion
          - option_space is not changed
```

This applies for all UTs

---

done